### ISSUE TITLE
Use autodiff in layers

### ISSUE THREAD

**Issue by JohannesNaegele** [JohannesNaegele/MaNN.jl#12]
Prepare general explanation of autodiff.
Why is backward diff used in neural nets?
See Wiki and https://juliadiff.org/
Probable best to use [Yota.jl](https://github.com/dfdx/Yota.jl) or [Zygote.jl](https://github.com/FluxML/Zygote.jl)
Implementation should do something similar to http://fluxml.ai/Flux.jl/stable/training/training/#man-training or
https://fluxml.ai/Flux.jl/stable/models/basics/#man-taking-gradients respectively
